### PR TITLE
Refactor MLE & ControlVariate strategies

### PR DESCRIPTION
**Pull request by bvdmitri**
This PR aims to resolve issue #29 by enhancing the documentation and refactoring the internal structure and naming conventions.

Notable changes include:
- The `buffer` argument is now part of `ControlVariateStrategy`.
- The `seed/rng` is now included in `ProjectionParameters` instead of strategies.
- `CVICostGradientObjective` has been renamed to `ProjectionCostGradientObjective`.
- Naming consistency has been improved, for example, by using `get_*` uniformly.

### CODE REVIEW

File: src/strategies/control_variate.jl [hunk 1]
```diff
@@ -9,66 +10,37 @@ A strategy for gradient descent optimization and gradients computations that res
...
-
-function with_state(strategy::ControlVariateStrategy, state)
-    return ControlVariateStrategy(
-        nsamples = getnsamples(strategy),
-        seed = getseed(strategy),
-        rng = getrng(strategy),
-        state = state,
-    )
+function getinitialpoint(
+    ::ControlVariateStrategy,
```
Comment (Nimrais) [inline src/strategies/control_variate.jl:31]: Why `getinitialpoint` depends on `strategy`? I think dispatch on strategy can be removed.

**Review by Nimrais — COMMENTED** [review]
I really like this refactoring, aside of small comment great work!

File: src/strategies/control_variate.jl (hunk 1 above)
Comment (bvdmitri) [reply src/strategies/control_variate.jl:31]: the idea here is that different strategies may provide different initial points

Comment (Nimrais) [reply src/strategies/control_variate.jl:31]: ok I see, because now it's same for MLE and ControlVariate, I haven't understood it.

Comment (bvdmitri) [reply src/strategies/control_variate.jl]: addressed

**Review by Nimrais — APPROVED** [review]

**Pull request merged**